libs/sdk-backend-tiger/src/convertors/toBackend/afm/TotalsConverter.ts

```diff
@@ -41,6 +51,125 @@ export function convertTotals(def: IExecutionDefinition): Total[] {
+     * For `Marginal Total`, in Tiger AFM, would need to iterate through both dimensions and obtain the missing totalDimensions items
+     * based on the attribute and column identifiers order in buckets.
+     */
+
+    const attributeBucket = bucketsFind(buckets, "attribute");
+    const columnBucket = bucketsFind(buckets, "columns");
+
+    const matchingAttributeBucket = attributeBucket ? bucketAttributes(attributeBucket) : [];
+    const matchingColumnBucket = columnBucket ? bucketAttributes(columnBucket) : [];
+
+    let newTotals = totals;
+
+    if (shouldAddGrandTotalOrMarginalTotal(attributeBucket, columnBucket)) {
+        const attributeIdentifiers = getAttributeBucketIdentifiers(matchingAttributeBucket);
+        const columnIdentifiers = getAttributeBucketIdentifiers(matchingColumnBucket);
+
+        attributeBucket!.totals?.forEach((attributeTotal, index) => {
+            let hasRowAndColumnGrandTotals = false;
+            let hasRowAndColumnSubTotals = false;
+            let attributeDimensionIndex = 0;
+            let columnDimensionIndex = 0;
+            columnBucket!.totals?.forEach((columnTotal, columnIndex) => {
+                const attributeMeasureId = attributeTotal.measureIdentifier;
+                const attributeType = attributeTotal.type;
+                const columnMeasureId = columnTotal.measureIdentifier;
+                const columnType = columnTotal.type;
+                // Check for totals from same measure and type
+                if (
+                    attributeAndColumnHasSameTotal(
+                        attributeMeasureId,
+                        attributeType,
+                        columnMeasureId,
+                        columnType,
+                    )
+                ) {
+                    // Check for grand totals rows/columns
+                    if (
+                        attributeTotal.attributeIdentifier === attributeIdentifiers[0] &&
+                        columnTotal.attributeIdentifier === columnIdentifiers[0]
+                    ) {
+                        hasRowAndColumnGrandTotals = true;
+                    }
+
+                    // Check for subtotals rows/columns
+                    if (
+                        attributeTotal.attributeIdentifier !== attributeIdentifiers[0] &&
+                        columnTotal.attributeIdentifier !== columnIdentifiers[0]
+                    ) {
+                        attributeDimensionIndex = attributeIdentifiers.findIndex(
+                            (aI) => aI === attributeTotal.attributeIdentifier,
+                        );
+                        columnDimensionIndex = columnIdentifiers.findIndex(
+                            (cI) => cI === columnTotal.attributeIdentifier,
+                        );
+
+                        hasRowAndColumnSubTotals = true;
+                    }
+                }
+
+                // extend marginal totals payload
```

Comment:
If possible, just try to explain in a few words how the slicing works for `totalDimensionItems` for "dim_0" and "dim_1".